src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -609,7 +609,7 @@ Block InterpreterSelectQuery::getSampleBlockImpl()
 
     query_info.query = query_ptr;
     query_info.has_window = query_analyzer->hasWindow();
-    if (storage)
+    if (storage && !options.only_analyze)
```

Comment:
Looks like this is the only changed line :)
As long as all test passed, it LGTM.